src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs

```diff
@@ -112,5 +112,22 @@ public static IEnumerable<ITypedElement> Navigate(this ITypedElement element, st
                 return element.Children(name);
             }
         }
+
+        public static string FpJoin(this IEnumerable<ITypedElement> collection, string separator)
+        {
+            //if the collection is empty return the empty result
+            if (!collection.Any())
+                return string.Empty;
+
+            //only join collections with string values inside
+            foreach (var typedElement in collection)
+            {
+                if (typedElement.InstanceType != "string")
```

Comment:
I would replace the foreach loop with this:

```
 if (!collection.All(c => c.Value is string))   
   throw Error.InvalidOperation("Join function can only be performed on string collections.");
```

Reply:
Thanks for the suggestion. I talked to ewout and replaced the impleentation with your suggestion.